subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/graph/LegendRenderer.java

```diff
@@ -33,15 +33,15 @@ public LegendRenderer(StyledTextOutput output) {
     public void printLegend() {
         if (hasConstraints) {
             output.println();
-            output.withStyle(Info).text("(c) - dependency constraint");
+            output.withStyle(Info).text("(c) - This element is a dependency constraint, not a dependency. Look for the matching dependency elsewhere in the tree.");
         }
         if (hasCyclicDependencies) {
             output.println();
-            output.withStyle(Info).println("(*) - dependencies omitted (listed previously)");
+            output.withStyle(Info).println("(*) - Repeat occurrence of a transitive dependency subtree. Gradle expands transitive dependency subtrees only once per project; repeat occurrences only display the root of the subtree, followed by this annotation.");
         }
         if (hasUnresolvableConfigurations) {
             output.println();
-            output.withStyle(Info).println("(n) - Not resolved (configuration is not meant to be resolved)");
+            output.withStyle(Info).println("(n) - A dependency or dependency configuration that cannot be resolved.");
```

Comment:
What does "in this context" mean here? I prefer not to equivocate, especially for a legend: the configuration either can be resolved, or it can't. In this case, it can't. Is there some context where it would be resolvable?